lib/net/dns/resolver.rb

```diff
@@ -1081,7 +1081,7 @@ def parse_config_file
         if RUBY_PLATFORM =~ /mswin32|cygwin|mingw|bccwin/
           require 'win32/resolv'
           arr = Win32::Resolv.get_resolv_info
-          self.domain = arr[0]
+          self.domain = arr[0]&.first
```

Comment:
I'm reading through the resolv.conf [man page](https://www.man7.org/linux/man-pages/man5/resolv.conf.5.html) and the following appears in the section for "search":

> The domain directive is an obsolete name for the search directive that handles one search list entry only.

I think based on that and what's going on here we actually want:

```suggestion
          self.searchlist = arr[0]
```

Reply:
Sounds good to me :+1:

@sfewer-r7 Happy to apply this change, or do you want us to apply it on ourselves and land it? :+1:

Reply:
Hi @adfoster-r7, yup that suggestions looks great (Thanks @smcintyre-r7!), happy to apply this as the best solution 👍